salt/modules/file.py

```diff
@@ -5448,15 +5448,22 @@ def check_file_meta(
 
     if contents is not None:
         # Write a tempfile with the static contents
-        tmp = salt.utils.files.mkstemp(
-            prefix=salt.utils.files.TEMPFILE_PREFIX, text=True
-        )
-        if salt.utils.platform.is_windows():
-            contents = os.linesep.join(
-                _splitlines_preserving_trailing_newline(contents)
+        if isinstance(contents, six.binary_type):
+            tmp = salt.utils.files.mkstemp(
+                prefix=salt.utils.files.TEMPFILE_PREFIX, text=False
+            )
+            with salt.utils.files.fopen(tmp, "wb") as tmp_:
+                tmp_.write(contents)
+        else:
+            tmp = salt.utils.files.mkstemp(
+                prefix=salt.utils.files.TEMPFILE_PREFIX, text=True
             )
-        with salt.utils.files.fopen(tmp, "w") as tmp_:
-            tmp_.write(salt.utils.stringutils.to_str(contents))
+            if salt.utils.platform.is_windows():
+                contents = os.linesep.join(
+                    _splitlines_preserving_trailing_newline(contents)
+                )
+            with salt.utils.files.fopen(tmp, "w") as tmp_:
+                tmp_.write(salt.utils.stringutils.to_str(contents))
```

Comment:
As for me it would look cleaner if we'd just set a `binary` flag and use it in the code.
```python
        text = not isinstance(contents, six.binary_type)
        tmp = salt.utils.files.mkstemp(
            prefix=salt.utils.files.TEMPFILE_PREFIX, text=text
        )
        if text:
            if salt.utils.platform.is_windows():
                contents = os.linesep.join(
                    _splitlines_preserving_trailing_newline(contents)
                )
            contents = salt.utils.stringutils.to_str(contents)
        with salt.utils.files.fopen(tmp, "w" + ("" if text else "b")) as tmp_:
            tmp_.write(contents)
```
This way we'll have less chance to lose business logic details in the future because we're not splitting the blocks.

Reply:
@smarsching please address @DmitryKuzmenko's comment in a follow up PR.